include/rfb.js

```diff
@@ -2122,6 +2175,219 @@ var RFB;
+            chromium_XT_scancode["F19"] = 0x0066;
+            chromium_XT_scancode["F20"] = 0x0067;
+            chromium_XT_scancode["F21"] = 0x0068;
+            chromium_XT_scancode["F22"] = 0x0069;
+            chromium_XT_scancode["F23"] = 0x006A;
+            chromium_XT_scancode["F24"] = 0x006B;
+            chromium_XT_scancode["Undo"] = 0xE008;
+            chromium_XT_scancode["Paste"] = 0xE00A;
+            chromium_XT_scancode["Cut"] = 0xE017;
+            chromium_XT_scancode["Copy"] = 0xE018;
+            chromium_XT_scancode["LaunchMail"] = 0xE01E;
+            chromium_XT_scancode["Eject"] = 0xE02C;
+            chromium_XT_scancode["Help"] = 0xE03B;
+
+            var gecko_XT_scancode = {};
+            gecko_XT_scancode["NumpadEqual"] = 0x0059;
+            gecko_XT_scancode["F13"] = 0x0064;
+            gecko_XT_scancode["F14"] = 0x0065;
+            gecko_XT_scancode["F15"] = 0x0066;
+            gecko_XT_scancode["F16"] = 0x0067;
+            gecko_XT_scancode["F17"] = 0x0068;
+            gecko_XT_scancode["F18"] = 0x0069;
+            gecko_XT_scancode["F19"] = 0x006A;
+            gecko_XT_scancode["F20"] = 0x006B;
+            gecko_XT_scancode["F21"] = 0x006C;
+            gecko_XT_scancode["F22"] = 0x006D;
+            gecko_XT_scancode["F23"] = 0x006E;
+            gecko_XT_scancode["F24"] = 0x0076;
+            gecko_XT_scancode["KanaMode"] = 0x0070;
+            gecko_XT_scancode["Lang2"] = 0x0071;
+            gecko_XT_scancode["Lang1"] = 0x0072;
+            gecko_XT_scancode["IntlRo"] = 0x0073;
+            gecko_XT_scancode["Convert"] = 0x0079;
+            gecko_XT_scancode["NonConvert"] = 0x007B;
+            gecko_XT_scancode["LaunchApp2"] = 0xE021;
+            gecko_XT_scancode["Power"] = 0xE05E;
+            gecko_XT_scancode["LaunchApp1"] = 0xE06B;
+            gecko_XT_scancode["LaunchMail"] = 0xE06C;
+            gecko_XT_scancode["MediaSelect"] = 0xE06D;
+
+            function checkIfRunningInSupportedBrowser() {
+                var brVersion = navigator.userAgent;
+
+                if (brVersion.indexOf("Chrome") !== -1 ||
+                    brVersion.indexOf("Chromium") !==-1) {
+
+                    return "Chrome";
+
+                } else if (brVersion.indexOf("Firefox") !== -1) {
+                    return "Firefox";
+
+                } else if (brVersion.indexOf("OPR") !== -1) {
+                    return "Opera";
+                }
+                return null;
+            }
+
+            this._FBU.rects--;
+
+            var supportedBrowser = checkIfRunningInSupportedBrowser();
+            if (supportedBrowser !== null) {
+                Util.Info("QEMU Key Event extension enabled");
+
+                this._QEMU_Extended_Key_Event = true;
+                this._keyboard.setQEMUVNCKeyboardHandler();
+
+                this._XT_scancode = common_XT_scancode;
+
+                if (supportedBrowser === "Chrome") {
+                    for (var code in chromium_XT_scancode) {
```

Comment:
this should be `Object.values` instead of `for ... in ..` so we don't have issues with people overriding the prototype incorrectly.